### PR TITLE
insights: add docs for using query-based contexts to filter insights

### PR DESCRIPTION
**Pull request by Joelkw**
Add docs for the v0 of supporting contexts as filters on your code insights. 

## Test plan

This is a docs update I have viewed/checked locally.

### CODE REVIEW

File: doc/code_insights/explanations/code_insights_filters.md [hunk 1]
```diff
@@ -18,6 +18,18 @@ For exclusion, only repositories that have a repository name **not** matching th
 
 If you combine both filters, the inclusion pattern will be applied first, then the exclusion pattern.
 
+### `context:` Query-based search context filters 
+
+You can use a [query-based search context](../../code_search/how-to/search_contexts.md#beta-query-based-search-contexts) to filter your insights to only results matching repositories that match the `repo:` filter of the query-based context.
```
Comment (coury-clark) [inline doc/code_insights/explanations/code_insights_filters.md:23]:
```suggestion
You can use a [query-based search context](../../code_search/how-to/search_contexts.md#beta-query-based-search-contexts) to filter your insights to only results matching repositories that match the `repo:` or `-repo:` filter of the query-based context. 
```

**Review by coury-clark — APPROVED** [review]
Just clarifying we can use negation! LGTM

File: doc/code_insights/explanations/code_insights_filters.md (hunk 1 above)
Comment (Joelkw) [reply doc/code_insights/explanations/code_insights_filters.md:23]: Good callout, thanks.

File: doc/code_insights/explanations/code_insights_filters.md [hunk 2]
```diff
@@ -18,6 +18,18 @@ For exclusion, only repositories that have a repository name **not** matching th
...
+
+You can use this to filter multiple insights to a group of repositories that need only be maintained in one location. When you update a context that's being used as a filter, the next time you load the page, the filtered insight will reflect the updated context. 
+
+At this time, all other filter keywords are not yet supported – only the `repo:` and `-repo:` keywords of a context are recognized. When creating your context, you can define any group of repos using the syntax `repo:(^github\.com/sourcegraph/sourcegraph$|^github\.com/sourcegraph/about$...)`. 
+
+### Setting more than one filter 
+
+Filters are intersected. This means if you use the search context `context:` filter that narrows your insight down from all repositories to some repo A, repo B, and repo C, and then you use the `repo:` filter that's set only to repo A, your insight will show only results from repo A. Similarly, if you were to use the same `context:` and then also use the `-repo:` exclusion filter set to repo C, your insight would show results from repo A and repo B.
```
Comment (AlicjaSuska) [inline doc/code_insights/explanations/code_insights_filters.md:31]: @Joelkw maybe we could also describe the case when:
- context is set (repos A, B and C)
- user adds repo: filter to include repo D - then repos A,B, C and D are included.

I wouldn't like users to get an impression that they can only inlude/exclude repos within the chosen context.

Comment (Joelkw) [reply doc/code_insights/explanations/code_insights_filters.md:31]: @AlicjaSuska I thought that was the case – that you can only include/exclude additional repos within the context (that's how search works, at least). I will confirm and update the docs if I am wrong!

**Pull request merged**